go-controller/pkg/ovn/base_network_controller_policy.go

```diff
@@ -659,75 +682,132 @@ func getPolicyType(policy *knet.NetworkPolicy) (bool, bool) {
 
 // getNewLocalPolicyPorts will find and return port info for every given pod obj, that is not found in
 // np.localPods.
-// if there are problems with fetching port info from logicalPortCache, pod will be added to errObjs.
-func (oc *DefaultNetworkController) getNewLocalPolicyPorts(np *networkPolicy,
-	objs ...interface{}) (policyPortsToUUIDs map[string]string, policyPortUUIDs []string, errObjs []interface{}) {
+// if there are problems with fetching port info from logicalPortCache, error will be added to returned error array.
+func (bnc *BaseNetworkController) getNewLocalPolicyPorts(np *networkPolicy,
```

Comment:
the reason why I made the current change is that for default network, GetPodNADToNetworkMapping() does not return the "default" NADName as expected. It may return real NAD name that is annoated in `v1.multus-cni.io/default-network`. That is why we usually need to have a xxxForNAD() function (see as addLogicalPortToNetwork())

Reply:
hmm, I don't find the code less clear. Also we'd like to make use of the same GetPodNADToNetworkMapping() anyway (not to invent another function doing things similiar), so I don't see how performance could be different.

Reply:
we have at least 2 duplicated code parts
```
on, networkMap, err := util.GetPodNADToNetworkMapping(pod, bnc.NetInfo)
		if err != nil || !on {
			if err != nil {
				// if we are not able to determine if this pod is on this network continue processing the remain pods
				// anyway, hopefully the pod will be handled in a future pod update event.
				klog.Warningf("Failed to determine if pod %s/%s is on network %s: %v",
					pod.Namespace, pod.Name, bnc.GetNetworkName(), err)
			}
			continue
		}
```
is called from `getNewLocalPolicyPorts` and `getExistingLocalPolicyPorts`
So I would like to have a single function that will do this part, and if you just add return "" for default network controller, then we can remove the second duplication which is:

```
portInfo, err := bnc.getNewLocalPolicyPortsForNAD(np, pod, types.DefaultNetworkName)
			if err != nil {
				errs = append(errs, err)
			} else if portInfo != nil {
				policyPortUUIDs = append(policyPortUUIDs, portInfo.uuid)
				policyPortsToUUIDs[portInfo.name] = portInfo.uuid
			}
			continue
```
is called in `getNewLocalPolicyPorts` twice: separately for default and secondary controllers.
So you can keep using `GetPodNADToNetworkMapping` from `bnc.getNADNames` and still have the code as I suggested before ^.

also
```
if !bnc.IsSecondary() {
		logicalPortName = util.GetLogicalPortName(pod.Namespace, pod.Name)
	} else {
		logicalPortName = util.GetSecondaryNetworkLogicalPortName(pod.Namespace, pod.Name, nadName)
	}
```
can be replaced with `bnc.GetLogicalPortName(pod, nadName)`, and let us have no extra `if !bnc.IsSecondary()` in the handlers